src/hooks/useModule.ts

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useRef } from "react";
+import content from "../content";
+import { Module, Section } from "../types";
+
+const useModule = (currentModule: Module) => {
+    const contentData = content[currentModule];
+    const totalPages = contentData.length - 1;
+    const totalBonusPages = contentData.filter(
+        (page) => page.section === Section.BonusContent
+    ).length;
+    const totalMainContentPages = totalPages - totalBonusPages;
+    const exampleTrajectoryPageNumber = contentData.findIndex(
```

Comment:
Should these be defined in the `useRef` definition (otherwise it's run every time) or in a `useMemo`?

It's also repeated from lines 23-31, so it seems like it could be refactored for DRY